packages/deployments/contracts/contracts/nomad-xapps/contracts/promise-router/PromiseRouter.sol

```diff
@@ -41,33 +63,63 @@ contract PromiseRouter is Version0, Router {
   // ======== Events =========
 
   /**
-   * @notice Emitted when a fees claim has been initialized in this domain
-   * @param domain The domain where to claim the fees
+   * @notice Emitted when a promise callback has been sent from this domain
+   * @param domain The domain where to execute the callback
    * @param remote Remote PromiseRouter address
    * @param transferId The transferId
    * @param callbackAddress The address of the callback
-   * @param data The calldata which will executed on the destination domain
+   * @param success The return success from the execution on the destination domain
+   * @param data The returnData from the execution on the destination domain
    * @param message The message sent to the destination domain
    */
-  event Send(uint32 domain, bytes32 remote, bytes32 transferId, address callbackAddress, bytes data, bytes message);
+  event Send(
+    uint32 domain,
+    bytes32 remote,
+    bytes32 transferId,
+    address callbackAddress,
+    bool success,
+    bytes data,
+    bytes message
+  );
 
   /**
-   * @notice Emitted when the a fees claim message has arrived to this domain
+   * @notice Emitted when a promise callback message has arrived to this domain
    * @param originAndNonce Domain where the transfer originated and the unique identifier
    * for the message from origin to destination, combined in a single field ((origin << 32) & nonce)
    * @param origin Domain where the transfer originated
    * @param transferId The transferId
    * @param callbackAddress The address of the callback
-   * @param data The calldata
+   * @param success The return success from the execution on the destination domain
+   * @param data The returnData from the execution on the destination domain
+   * @param message The message sent to the destination domain
    */
   event Receive(
     uint64 indexed originAndNonce,
     uint32 indexed origin,
     bytes32 transferId,
     address callbackAddress,
-    bytes data
+    bool success,
+    bytes data,
+    bytes message
```

Comment:
is this message (informally) bounded in length somewhere? if not, is emitting the message advisable?

Reply:
there are upper bounds within the nomad system when you send a message, so we kind of offload that to them